tests/entrypoints/openai/test_truncation.py

```
@@ -73,17 +73,11 @@ async def test_zero_truncation_size(client: openai.AsyncOpenAI):
         "truncate_prompt_tokens": truncation_size
     }
 
-    with pytest.raises(openai.BadRequestError) as err:
-        await client.post(path="embeddings", cast_to=object, body={**kwargs})
-
-    assert err.value.status_code == 400
-    error_details = err.value.response.json()["error"]
+    response = await client.post(path="embeddings",
```

Doesn't this mean that the behavior has changed?

---

Yes it does have changed since truncate_prompt_tokens is used inconsistently across the code base.
When truncate_prompt_tokens = 0:
1. for token id only input, it does truncate
2. for text input, it doesn't truncate, it silently ignore as HF tokenizer ignores it when the length is 0
In renderer, I tried to unify the behaviour by doing an early return: https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/renderer.py#L114, so both cases are handled the same way.